build/accelerate_launch.py

```diff
@@ -57,23 +57,27 @@ def main():
 
     parser = launch_command_parser()
     # determine which flags are store true actions (don't require a value to be set)
-    store_action_params = {}
-    for action in parser._actions:
-        if type(action).__name__ == "_StoreTrueAction":
-            store_action_params[action.dest] = action
+    actions_type_map = {
+        action.dest: type(action).__name__ for action in parser._actions
+    }
 
     # parse accelerate_launch_args
     accelerate_launch_args = []
     accelerate_config = json_configs.get("accelerate_launch_args", {})
     if accelerate_config:
         logging.info("Using accelerate_launch_args configs: %s", accelerate_config)
         for key, val in accelerate_config.items():
-            # For flags that don't have value, ie. --quiet, only add if value is true
-            if store_action_params.get(key) and val:
-                accelerate_launch_args.append(f"--{key}")
+            if actions_type_map.get(key) == "_AppendAction":
+                for param_val in val:
+                    accelerate_launch_args.extend([f"--{key}", str(param_val)])
             else:
                 accelerate_launch_args.append(f"--{key}")
-                accelerate_launch_args.append(str(val))
+                # Only need to add key for params that aren't flags ie. --quiet
+                if actions_type_map.get(key) not in [
+                    "_StoreTrueAction",
+                    "_StoreFalseAction",
+                ]:
+                    accelerate_launch_args.append(str(val))
```

Comment:
Can you flip this? So that we test if it is StoreActionTrue or StoreActionFalse and if not append both key and value.

I was looking into the params and there are 5 types for this arg parser: `{'_HelpAction', '_AppendAction', '_StoreTrueAction', '_StoreConstAction', '_StoreAction'}`

_HelpAction should technically go with _StoreTrueAction. 

I actually don't see _StoreFalseAction here. This is because there is only one of these args currently `--no_tpu_cluster` and there is also the opposite `--tpu_cluster` and they both assign to the same var/dest `tpu_use_cluster`. So I agree its good to check for both but just an interesting detail. Interstingly though `tpu_use_cluster` is of action type _StoreConstAction.

_StoreAction is the main one we want, not _AppendAction which looks to be for lists. These are in teh same category of how we handle the flag

Reply:
My understanding is you are doing if AppendAction do `--flag value` else do `--flag`. But we want to default to doing `--flag value` as there are many more of these. For example `num_processes` is _StoreAction which means it should be added as  `--flag value`. But the way your logic works it would be added as `--flag`. Lists are actually quite the headache...

Reply:
Thus instead of checking for `if actions_type_map.get(key) == "_AppendAction":` you should be checking for `if actions_type_map.get(key) == "_StoreAction":` at minimum

Reply:
But I see your point that altho this arg parser has a lot of the StoreAction type, the type of actions are more likely to not need to be flag and valut but just flag. Here's a test arg parser I wrote that has all of the action types
```
usage: test_args.py [-h] [--store STORE] [--constant] [--true] [--false] [--append APPEND]
                    [--append_const] [--verbose] [--version] [--extend EXTEND [EXTEND ...]]
```

So the store, append, and extend ones are the only ones where you pass in lag and value. So should we check for those in initial if statement adn then do else for the rest?